### PR TITLE
fix(windows): add x64 build fix invalid memory access

### PR DESCRIPTION
**Pull request by rc-swag**
This commit adds x64 build for the imsample.dll.
It also fixes the a invalid memory access error that
occured when UnloadRules was called when a Keyboard
had been created but the rules not yet loadeded.

KeymanIMReloadConfig function returned an invalid
code execution point this was fixed by adding it
to the imsample.def

Will skip testing as this is a sample keyboard and the tester had more than enough to do.
@keymanapp-test-bot skip

### CODE REVIEW

File: windows/src/developer/samples/imsample/imsample.cpp [hunk 1]
```diff
@@ -657,7 +657,7 @@ extern "C" BOOL __declspec(dllexport) WINAPI KeymanIMConfigure(PSTR keyboardname
 void WriteRegSetting(PSTR text, int value)
 {
 	HKEY hkey;
-	if(RegOpenKeyEx(HKEY_CURRENT_USER, "Software\\Tavultesoft\\Keyman\\5.0", 
+  if (RegOpenKeyEx(HKEY_CURRENT_USER, "Software\\Keyman\\Keyman Engine\\Active Keyboards\\impsample",
```
Comment (mcdurdin) [inline windows/src/developer/samples/imsample/imsample.cpp:660]:
```suggestion
  if (RegOpenKeyEx(HKEY_CURRENT_USER, "Software\\Keyman\\Keyman Engine\\Active Keyboards\\imsample", 
```

File: windows/src/developer/samples/imsample/imsample.cpp [hunk 2]
```diff
@@ -668,7 +668,7 @@ void WriteRegSetting(PSTR text, int value)
 int ReadRegSetting(PSTR text)
 {
 	HKEY hkey;
-	if(RegOpenKeyEx(HKEY_CURRENT_USER, "Software\\Tavultesoft\\Keyman\\5.0", 
+	if(RegOpenKeyEx(HKEY_CURRENT_USER, "Software\\Keyman\\Keyman Engine\\Active Keyboards\\impsample",
```
Comment (mcdurdin) [inline windows/src/developer/samples/imsample/imsample.cpp:671]:
```suggestion
	if(RegOpenKeyEx(HKEY_CURRENT_USER, "Software\\Keyman\\Keyman Engine\\Active Keyboards\\imsample", 
```

File: windows/src/developer/samples/imsample/imsample.cpp [hunk 3]
```diff
@@ -791,7 +791,9 @@ PWSTR extstr(char *p)
 void UnloadRules(PSTR KeyboardName)
 {
 	int nkbi;
-
+  if (nkeyboards == 0) {
+    return;
+  }
```
[the remark this answers is not in the document]
Comment (rc-swag) [reply windows/src/developer/samples/imsample/imsample.cpp:796]: Yes you are correct. I'm not sure what I was thinking?? for some reason I had in my mind that the zero nkeyboard case would have nkbi = 1; but clearly, that will not be the case.

Comment (rc-swag) [inline windows/src/developer/samples/imsample/imsample.cpp:796]:
```suggestion
```

**Review by mcdurdin — APPROVED** [review]
LGTM

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.176-alpha](https://keyman.com/downloads/releases/alpha/15.0.176)

**Pull request merged**